processor-api/src/main/java/com/synopsys/integration/alert/processor/api/filter/extractor/VulnerabilityNotificationExtractor.java

```diff
@@ -0,0 +1,94 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.processor.api.filter.extractor;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.google.gson.Gson;
+import com.synopsys.integration.alert.common.rest.model.AlertNotificationModel;
+import com.synopsys.integration.alert.processor.api.filter.model.FilterableNotificationWrapper;
+import com.synopsys.integration.blackduck.api.generated.enumeration.VulnerabilitySeverityType;
+import com.synopsys.integration.blackduck.api.manual.component.AffectedProjectVersion;
+import com.synopsys.integration.blackduck.api.manual.component.VulnerabilityNotificationContent;
+import com.synopsys.integration.blackduck.api.manual.component.VulnerabilitySourceQualifiedId;
+import com.synopsys.integration.blackduck.api.manual.enumeration.NotificationType;
+
+@Component
+public class VulnerabilityNotificationExtractor extends NotificationExtractor {
+    private Gson gson;
+
+    @Autowired
+    public VulnerabilityNotificationExtractor(Gson gson) {
+        super(NotificationType.VULNERABILITY);
+        this.gson = gson;
+    }
+
+    @Override
+    public FilterableNotificationWrapper convertToFilterableNotificationWrapper(AlertNotificationModel alertNotificationModel) {
+        VulnerabilityNotificationContent vulnerabilityNotificationContent = gson.fromJson(alertNotificationModel.getContent(), VulnerabilityNotificationContent.class);
+
+        List<String> affectedProjectNames = vulnerabilityNotificationContent.getAffectedProjectVersions()
+                                                .stream()
+                                                .map(AffectedProjectVersion::getProjectName)
+                                                .collect(Collectors.toList());
+        List<String> applicableSeverityTypes = getApplicableSeverityTypes(vulnerabilityNotificationContent)
+                                                   .stream()
+                                                   .map(VulnerabilitySeverityType::name)
+                                                   .collect(Collectors.toList());
+
+        return FilterableNotificationWrapper.vulnerability(alertNotificationModel, vulnerabilityNotificationContent, affectedProjectNames, applicableSeverityTypes);
+    }
+
+    private Set<VulnerabilitySeverityType> getApplicableSeverityTypes(VulnerabilityNotificationContent vulnerabilityNotificationViewContent) {
```

Comment:
Why not have this return the names so you don't have to have the extra streaming on the `Set` this method returns?

Reply:
I actually need to convert this to a list later on, so I will need to stream it afterwards regardless. We could maybe convert the List<String> for policyNames and vulnSeverities to Set<String> As it's used for filtering and we don't want duplication anyway

Reply:
You don't need to stream. You can use a constructor to create the list out of it. Or you could call `Stream.unique()` and then collect to a list.